### PR TITLE
palindromic partitions yo

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = int_fast64_t;
+
+const ll P = 69421, M = 1e9 + 9;
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:23]: different lines

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+The greedy approach is implemented below. We will loop from left to right, adding characters one at a time until the left string is the same as the right string, in which we can create two partitions. To check the equality of these strings, we will use string hashing.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = int_fast64_t;
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:21]: what

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const ll P = 69421, M = 1e9 + 9;
+
+int main(){
+	cin.tie(0) -> sync_with_stdio(0);
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:26]: remove if runs w/o

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+using namespace std;
+using ll = int_fast64_t;
+
+const ll P = 69421, M = 1e9 + 9;
+
+int main(){
+	cin.tie(0) -> sync_with_stdio(0);	
+	int t; 
+	cin >> t;
+	goto outer;
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:29]: what

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	
+	outer:
+	if(t-- == 0) {
+		return 0;
+	}
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:34]: envy have you heard of a loop

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+			// we have taken all the characters
+			if (start * 2 == n) {
+				cout << ans << endl;
+			}
+			else {
+				cout << ans + 1 << endl;
+			}
+			goto outer;
+		}
+		else {
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:65]:
```suggestion
		} else {
```
same for others

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
weird

File: solutions/ceoi-17-PalindromicPartitions.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/ceoi-17-PalindromicPartitions.mdx:21]: what?

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+source: CEOI
+title: 2017 - Palindromic Partitions
+author: Chongtian Ma
+---
+
+[Official Analysis](https://csacademy.com/contest/ceoi-2017-day-2/task/palindromic-partitions/solution/)
+
+## Implementation
+
+The greedy approach is implemented below. We will loop from left to right, adding characters one at a time until the left string is the same as the right string, in which we can create two partitions. To check the equality of these strings, we will use string hashing.
```
Comment (SansPapyrus683) [inline solutions/ceoi-17-PalindromicPartitions.mdx:12]: adding characters to what?

Comment (envyaims) [reply solutions/ceoi-17-PalindromicPartitions.mdx:12]: to both strings? what else?

Comment (SansPapyrus683) [reply solutions/ceoi-17-PalindromicPartitions.mdx:12]: wdym "both strings"?
don't expect the reader to automatically know what variables you're using

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+using namespace std;
+using ll = long long;
+
+const ll P = 69421;
+const ll M = 1e9 + 9;
+
+int main(){
+	int t; 
+	cin >> t;
+	while (t--) {
```
Comment (ryanchou-dev) [inline solutions/ceoi-17-PalindromicPartitions.mdx:29]:
```suggestion
	int tc; 
	cin >> tc;
	while (tc--) {
```
optional

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const ll P = 69421;
```
Comment (ryanchou-dev) [inline solutions/ceoi-17-PalindromicPartitions.mdx:23]: brilliant hash

**Review by ryanchou-dev — APPROVED** [review]
image

File: solutions/ceoi-17-PalindromicPartitions.mdx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+title: 2017 - Palindromic Partitions
+author: Chongtian Ma
+---
+
+[Official Analysis](https://csacademy.com/contest/ceoi-2017-day-2/task/palindromic-partitions/solution/)
+
+## Implementation
+
+The greedy approach is implemented below. We are going to initialize two strings (in this case, hash values) - one built with the $i'$th character and the other with its $n - i - 1$'th complement. We will loop from left to right, adding characters one at a time to their respective strings until the left string is the same as the right string, in which we can create two partitions. To check the equality of these strings, we will use string hashing.
+
```
Comment (ryanchou-dev) [inline solutions/ceoi-17-PalindromicPartitions.mdx:13]: add time complexity?

File: solutions/ceoi-17-PalindromicPartitions.mdx (hunk 9 above)
Comment (envyaims) [reply solutions/ceoi-17-PalindromicPartitions.mdx:22]: exactly uwu

**Review by SansPapyrus683 — APPROVED** [review]
we got 'em

**Pull request merged**